### PR TITLE
Fix createJavadoc failing issue

### LINKED ISSUE #24631: Javadoc Creation Fails
**Description:**
The Javadoc creation fails.

**Steps to reproduce:**
Execute the commands `./gradlew :ballerina-parser:createJavadoc` and `./gradlew :ballerina-lang:createJavadoc` or build without skipping the checks using `./gradlew build -x test -x :composer-library:npmInstall -x :composer-library:npmBuild`

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #24631

https://github.com/ballerina-platform/ballerina-lang/blob/13709b8795480a76672aae0bb143e06a641973bc/compiler/ballerina-lang/src/main/java/org/ballerinalang/project/Project.java#L38
and
https://github.com/ballerina-platform/ballerina-lang/blob/13709b8795480a76672aae0bb143e06a641973bc/compiler/ballerina-lang/src/main/java/org/ballerinalang/project/Project.java#L46

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/project/Project.java [hunk 1]
```diff
@@ -35,15 +35,15 @@
      * Returns true if the module exists in the project.
      *
      * @param moduleId Module Id
-     * @return
+     * @return boolean
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/java/org/ballerinalang/project/Project.java:38]: In java doc comments the @returns specify what it returns not the type of return value, the type of the return value is alredy in the function signature.

In this instance it's something like `module exists in the project`

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java [hunk 2]
```diff
@@ -478,7 +478,7 @@ public Result(ArrayDeque<Solution> fixes, int matches) {
      *
      * @since 1.2.0
      */
-    enum Action {
+    public enum Action {
```
Comment (rdhananjaya) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java:481]: This doesn't looks like a cause for java doc issue. Do you have the log for this?

Comment (sanjana) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java:481]: Yes, The `./gradlew :ballerina-parser:createJavadoc` command produces the following warnings 

- `/path/to/repo/ballerina-lang/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java:468: warning - Tag @link: reference not found: Action#KEEP`

- `/path/to/repo/ballerina-lang/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/AbstractParserErrorHandler.java:424: warning - Tag @link: reference not found: Action`

and 
`> Task :ballerina-parser:createJavadoc FAILED`
`FAILURE: Build failed with an exception.`

File: compiler/ballerina-lang/src/main/java/org/ballerinalang/project/Project.java (hunk 1 above)
Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/org/ballerinalang/project/Project.java]: Updated the Javadocs.

**Pull request merged**